tests/linen/linen_test.py

```diff
@@ -146,6 +146,21 @@ def test_group_norm_raises(self):
     with self.assertRaises(ValueError):
       model_cls.init_with_output(key2, x)
 
+  def test_batch_norm_multi_init(self):
+    class Foo(nn.Module):
+      @nn.compact
+      def __call__(self, x):
+        norm = nn.BatchNorm(
+            name="norm",
+            use_running_average=False,
+            axis_name="batch",
+        )
+        x = norm(x)
+        return norm(x)
+
+    key = random.PRNGKey(0)
+    model = Foo()
+    variables = model.init(key, jnp.ones((10,)))
```

Comment:
This test only checks that no error is thrown. Can we also test that the return values are correct? It sounds like a simple reasonable check is that the two calls to norm() return the same thing (I think?) so you could return them both from Foo.__call__ (`return norm(x), norm(x)`) and then just assert that you get same arrays out

Reply:
Done